### PR TITLE
NFDIV-2356 : Joint CO applied notifications

### PR DESCRIPTION
**Pull request by sunilnatikar**
### Change description ###
Notifications added for joint CO submissions

### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-2356

**Before merging a pull request make sure that:**

- [x] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [x] this ticket been reviewed by QA
- [x] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java [hunk 1]
```diff
@@ -22,23 +26,49 @@
 
     @Override
     public void sendToApplicant1(final CaseData caseData, final Long id) {
-        log.info("Notifying applicant 1 that their conditional order application has been submitted: {}", id);
+
+        EmailTemplateName templateName;
+
+        if (caseData.getApplicationType().isSole()) {
+            log.info("Notifying applicant that their conditional order application has been submitted: {}", id);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java:33]: worth logging template being used

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java [hunk 2]
```diff
@@ -22,23 +26,49 @@
...
+            templateName = CITIZEN_APPLIED_FOR_CONDITIONAL_ORDER;
+        } else if (alreadyApplied(caseData, APPLICANT2)) {
+            log.info("Notifying applicant 1 that both applicants has submitted their conditional order application: {}", id);
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java:36]:
```suggestion
            log.info("Notifying applicant 1 that both applicants have submitted their conditional order application: {}", id);
```

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java [hunk 3]
```diff
@@ -22,23 +26,49 @@
...
     public void sendToApplicant2(final CaseData caseData, final Long id) {
-        if (!caseData.getApplicationType().isSole() && !alreadyApplied(caseData, APPLICANT2)) {
-            log.info("Notifying applicant 2 that their partner has submitted a conditional order application: {}", id);
+        if (!caseData.getApplicationType().isSole()) {
+
+            EmailTemplateName templateName;
+            Map<String, String> templateMap;
+
+            if (alreadyApplied(caseData, APPLICANT2)) {
+                log.info("Notifying applicant 2 that both applicants has submitted their conditional order application: {}", id);
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java:59]:
```suggestion
                log.info("Notifying applicant 2 that both applicants have submitted their conditional order application: {}", id);
```

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java [hunk 4]
```diff
@@ -21,23 +25,44 @@
 
     @Override
     public void sendToApplicant1(final CaseData caseData, final Long id) {
-        if (!alreadyApplied(caseData, APPLICANT1)) {
+
+        EmailTemplateName templateName;
+        Map<String, String> templateMap;
+
+        if (alreadyApplied(caseData, APPLICANT1)) {
+            log.info("Notifying applicant 1 that both applicants has submitted their conditional order application: {}", id);
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java:33]:
```suggestion
            log.info("Notifying applicant 1 that both applicants have submitted their conditional order applications: {}", id);
```

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java [hunk 5]
```diff
@@ -21,23 +25,44 @@
...
     }
 
     @Override
     public void sendToApplicant2(final CaseData caseData, final Long id) {
-        log.info("Notifying applicant 2 that their conditional order application has been submitted: {}", id);
+
+        EmailTemplateName templateName;
+
+        if (alreadyApplied(caseData, APPLICANT1)) {
+            log.info("Notifying applicant 2 that both applicants has submitted their conditional order application: {}", id);
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java:56]:
```suggestion
            log.info("Notifying applicant 2 that both applicants have submitted their conditional order applications: {}", id);
```

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java (hunk 1 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java]: done

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java (hunk 2 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java]: corrected

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java (hunk 3 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant1AppliedForConditionalOrderNotification.java]: corrected

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java (hunk 4 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java]: corrected

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java (hunk 5 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/Applicant2AppliedForConditionalOrderNotification.java]: corrected

**Pull request merged**